### DISCUSSION TITLE
[Feature]使用Waline管理帖子和评论

### DISCUSSION THREAD

**Discussion by Buer-Nahida** [share121/inter-knot#1416] [其他]
# Raw Issue at #1414

## Q: 为什么我要提出这个Issue

绳网跨域助手+Github的管理方式很有局限性：

1. 由于是Github Discussions提供的发帖方式，管理方式都得跟着Github走，有可能因为滥用资源被封（参考https://github.com/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee/issues/603
）
2. 发帖人无法删帖（原因同上）
3. Github在国内的半墙状态和手机浏览器没几个支持插件导致很多想上绳网的用户被拦在外边（或许是好处？）
4. 访问速度慢（原因同第一条）

## Q: 什么是Waline

[Waline](https://waline.js.org)是一款一款简洁、安全的评论系统，支持发送图片的操作，且Vercel（负责前端和评论系统）+TiDB（负责Waline数据库）这个部署方式完全免费（TiDB应该是超过多少数据后收费，毕竟数据中心的开销是一个巨大的数字）

[我的个人网站](https://naxida.me)就使用了这个评论系统，所以我体会过它的部署难度，它部署起来很简单。

虽然Waline基本被用于博客的评论系统，但它支持发送图片，也支持回复，所以可以把每个评论归类为一条帖子

优点：
- 本地部署，不被Github牵着鼻子走
- 没有10M图片的限制
- 不怕被封
- Vercel和TiDB免费提供域名
- 用户登录门槛低，不用再找魔法、安装油猴插件、安装油猴脚本了
- 速度或许会快一点

缺点：
- 部署方式虽简单，但相关代码的修改是一大工程
- 可能会让更多的小鬼进来污染环境
- ***可能***不支持视频

## Q: 我可以贡献这个功能吗？

我不能贡献一个写好的功能，因为Waline***需要您自己部署***，况且我还是刚入门前端的萌新，而且我是个***初二***的学生，开学后就***初三***了，所以我***没有时间***把代码写好创建一个Pull request来把代码提交上去，但我可以参与贡献。

**Comment by match-fire** [comment]
建了，很好用

**Comment by match-fire** [comment]
但我套个CDN会导致重定向次数过多，不知道为啥
![image](https://github.com/user-attachments/assets/5975e370-4a40-4147-bada-9ad5bb53390f)

**Comment by user1-cloud** [comment]
好家伙才初二吗😮
不过推倒底层架构重做的工作量还是**太大了**

**Comment by ghost** [comment]
个人感觉用GitHub作为门槛真的是好处，没必要换Waline（ZZZ原作中绳网也不是随便什么人能上的东西）
然后可以做的我觉得就是写个UI不用跳转到GitHub Discussion来写东西

**Comment by capacitor1** [comment]
> Reply::个人感觉用GitHub作为门槛真的是好处，没必要换Waline...

我认为，如果不用git的来写的话，还是得另加数据库，不然写的东西咋保存🤔
